### ISSUE TITLE
Syncbase docs: Dart API docs

### ISSUE THREAD

**Issue by aghassemi** [vanadium/issues#808]
- [ ] Setup for generating and publishing Dart documentation.
- [ ] Actually documenting the code.
- [ ] Ensuring mojom generated types are somehow documented as well.
- [ ] Referencing the hosted API doc from the website

**Comment by afandria** [comment]
For the last one, we should ask (or send a pull request to) Mojo to carry over the .mojom file's comments the way that the vdl tool does. This would also let us place our own comment block at the header of the generated file, instead of theirs.

**Comment by aghassemi** [comment]
yeah one issue we ran with this approach in VDL was different languages have different documentation requirements and styles. For example, Go does not care about documenting parameters but Dart and JS do, so should we be forcing the user to document params anyway in VDL/mojom eventhough the language they compile to does not have that notion? Also there could be conflicting guidelines, like Go docs start with the name of the method, but many other languages don't. I believe @tatatodd has more issues related to it. I don't know if we can do a documentation style that can be translated into documentation for all supported languages. It would definitely be nice though!
